jni/include/jni_util.h

```diff
@@ -200,13 +204,36 @@ namespace knn_jni {
         jfieldID GetFieldID(JNIEnv * env, jclass clazz, const char *name, const char *sig) final;
         jint CallNonvirtualIntMethodA(JNIEnv *env, jobject obj, jclass clazz, jmethodID methodID, jvalue *args) final;
         jlong CallNonvirtualLongMethodA(JNIEnv * env, jobject obj, jclass clazz, jmethodID methodID, jvalue* args) final;
+        void CallNonvirtualVoidMethodA(JNIEnv * env, jobject obj, jclass clazz, jmethodID methodID, jvalue* args) final;
         void * GetPrimitiveArrayCritical(JNIEnv * env, jarray array, jboolean *isCopy) final;
         void ReleasePrimitiveArrayCritical(JNIEnv * env, jarray array, void *carray, jint mode) final;
 
     private:
         std::unordered_map<std::string, jclass> cachedClasses;
         std::unordered_map<std::string, jmethodID> cachedMethods;
-    };
+    };  // class JNIUtil
+
+    /**
+     * It's common cleaner to release a primitive array within its destructor.
+     * Ex: JNIReleaseElements release_int_array_elements {[=](){
+     *   jniUtil->ReleaseIntArrayElements(env, idsJ, idsCpp, JNI_ABORT);
+     * }};
+     */
+    struct JNIReleaseElements {
+      explicit JNIReleaseElements(std::function<void()> _release_func)
+          : release_func(std::move(_release_func)) {
+      }
+
+      ~JNIReleaseElements() {
+        try {
+          release_func();
```

Comment:
Can we ensure  release_func is a valid function before calling it to prevent unexpected behavior.

Reply:
will update in the next rev